packages/babel-register/src/node.js

```diff
@@ -129,20 +111,27 @@ function hookExtensions(_exts) {
   });
 }
 
-hookExtensions(util.canCompile.EXTENSIONS);
+hookExtensions([".js", ".jsx", ".es6", ".es"]);
 
 export default function (opts?: Object = {}) {
-  if (opts.only != null) only = util.arrayify(opts.only, util.regexify);
-  if (opts.ignore != null) ignore = util.arrayify(opts.ignore, util.regexify);
-
-  if (opts.extensions) hookExtensions(util.arrayify(opts.extensions));
+  if (opts.extensions) hookExtensions(opts.extensions);
 
   if (opts.cache === false) cache = null;
 
   delete opts.extensions;
-  delete opts.ignore;
   delete opts.cache;
```

Comment:
Sorry for commenting on an old thread but is there any reason you are not using [`lodash.omit`](https://lodash.com/docs/4.17.4#omit)? [Another instance](/babel/babel/blob/7.0/packages/babel-cli/src/babel/index.js#L133-L140)